wgsl/index.bs

```diff
@@ -4242,6 +4250,69 @@ TODO(dsinclair): Need gather operations
 
 ## Atomic built-in functions ## {#atomic-builtin-functions}
 
+## Data packing functions ## {#pack-functions}
+
+<table class='data'>
+  <thead>
+    <tr><th>Built-in<th>Description
+  </thead>
+  <tr algorithm="packing 4x8snorm">
+    <td class="nowrap">`pack4x8snorm`(|e|: vec4&lt;f32&gt;) -> u32
+    <td>Converts four normalized floating point values to 8-bit signed integers, and then combines them
+        into one `u32` value.<br>
+        Component |e|[|i|] of the input is converted to an 8-bit twos complement integer value
+        &lfloor; 0.5 + 127 &times; min(1, max(-1, |e|[|i|])) &rfloor; which is then placed in bits
+        8 &times; |i| through 
+        8 &times; |i| + 7 of the result.
+
+  <tr algorithm="packing 4x8unorm">
+    <td class="nowrap">`pack4x8unorm`(|e|: vec4&lt;f32&gt;) -> u32
+    <td>Converts four normalized floating point values to 8-bit unsigned integers, and then combines them
+        into one `u32` value.<br>
+        Component |e|[|i|] of the input is converted to an 8-bit unsigned integer value
+        &lfloor; 0.5 + 255 &times; min(1, max(0, |e|[|i|])) &rfloor; which is then placed in bits
+        8 &times; |i| through 
+        8 &times; |i| + 7 of the result.
+
+  <tr algorithm="packing 2x16snorm">
+    <td class="nowrap">`pack2x16snorm`(|e|: vec2&lt;f32&gt;) -> u32
+    <td>Converts two normalized floating point values to 16-bit signed integers, and then combines them
+        into one `u32` value.<br>
+        Component |e|[|i|] of the input is converted to a 16-bit twos complement integer value
+        &lfloor; 0.5 + 32767 &times; min(1, max(-1, |e|[|i|])) &rfloor; which is then placed in bits
+        16 &times; |i| through 
+        16 &times; |i| + 15 of the result.
+
+  <tr algorithm="packing 2x16unorm">
+    <td class="nowrap">`pack2x16unorm`(|e|: vec2&lt;f32&gt;) -> u32
+    <td>Converts two normalized floating point values to 8-bit unsigned integers, and then combines them
+        into one `u32` value.<br>
+        Component |e|[|i|] of the input is converted to a 16-bit unsigned integer value
+        &lfloor; 0.5 + 65535 &times; min(1, max(0, |e|[|i|])) &rfloor; which is then placed in bits
+        16 &times; |i| through 
+        16 &times; |i| + 15 of the result.
+
+  <tr algorithm="packing 2xf16">
+    <td class="nowrap">`pack2xf16`(|e|: vec2&lt;f32&gt;) -> u32
```

Comment:
This ones name seems to be the odd one out. Either this should be `pack2x16float` to match the above, or could we change the above to be `pack4xu8`, `pack2xu16`, `pack2xi8`, `pack2xi16` which matches the `pack2xf16`. In theory, if we get the smaller types, then those names should all match.

Reply:
These pack (and later unpack) functions are for doing in value types what happens when reading/writing texels from storage textures.  So I wanted to match the texel channels. 

So you're right it should be pack2x16float.  I'll fix it.

Reply:
Done